pkg/utils/urls/replace.go

```diff
@@ -21,5 +21,25 @@ func ReplaceHost(orgURL, host string) string {
 		u.Scheme = ""
 	}
 	u.Host = host
-	return strings.TrimPrefix(u.String(), "//")
+	return strings.ReplaceAll(strings.TrimPrefix(u.String(), "//"), "%2F", "/")
+}
+
+// ToAPIEndpoint turns URL to a valid API endpoint used in
+// a containerd config file for a local registry.
+func ToAPIEndpoint(URL string) string {
+	index := strings.Index(URL, "/")
+	if index == -1 {
+		return URL
+	}
+	return URL[:index] + "/v2" + URL[index:]
+}
+
+// ToAPIEndpoints utilizes ToAPIEndpoint to turn all URLs from a
+// map to valid API endpoints for a local registry.
+func ToAPIEndpoints(URLs map[string]string) map[string]string {
```

Comment:
Why is this a map and not a slice?

Reply:
If we make it as a slice, the relationship between which position in the slice and which registry that position maps to is implicit, thus resulting in some maintenance issue. Whereas, map turns that relationship to be explicit.